.github/workflows/deploy.yml

```diff
@@ -0,0 +1,70 @@
+name: Deploy to Amazon ECS
+
+on:
+  release:
+    types: [created, edited, published]
+  push: # temporary for testing
+  workflow_dispatch: # temporary for testing
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read # This is required for actions/checkout
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@b5ca514318bd6ebac0fb2aedd5d36ec1b5c232a2 #v3.10.0
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@ececac1a45f3b08a01d2dd070d28d111c5fe6722 #v4.1.0
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-duration-seconds: 1200
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@062b18b96a7aff071d4dc91bc00c4c1a7945b076 #v2.0.1
+
+      - name: Generate image name
+        id: generate-img-name
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          # IMAGE_TAG: ${{ github.sha }}
+          IMAGE_TAG: latest
+          IMAGE_NAME: safe-watcher
+        run: |
+          echo "image-name=${{ env.ECR_REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.IMAGE_TAG }}" \
+          >> $GITHUB_OUTPUT
+
+      - name: Build, tag and push image to Amazon ECR
```

Comment:
If you split the `push` part out, then you can build and tag, then log into AWS and finally do the push. This way, there are no AWS-related environment variables during the build.